cumulusci/tasks/bulkdata/step.py

```diff
@@ -281,12 +291,48 @@ def load_records(self, records):
 
         for count, csv_batch in enumerate(self._batch(records)):
             self.context.logger.info(f"Uploading batch {count + 1}")
-            self.batch_ids.append(self.bulk.post_batch(self.job_id, csv_batch))
-
-    def _batch(self, records):
-        """Return a generator of generators, where each child generator is batched."""
-        for batch in batch_iterator(records, self.api_options.get("batch_size", 10000)):
-            yield self._csv_generator(batch)
+            self.batch_ids.append(self.bulk.post_batch(self.job_id, iter(csv_batch)))
+
+    def _batch(self, records, n=10000, char_limit=10000000):
+        serialized_csv_fields = self._serialize_csv_record(self.fields)
+        len_csv_fields = len(serialized_csv_fields)
+
+        # append fields to first row
+        batch = [serialized_csv_fields]
+        current_chars = len_csv_fields
+        for record in records:
+            serialized_record = self._serialize_csv_record(record)
+            # Does the next record put us over the character limit?
+            if len(serialized_record) + current_chars > char_limit:
+                yield batch
+                batch = [serialized_csv_fields]
+                current_chars = len_csv_fields
+
+            batch.append(serialized_record)
+            current_chars += len(serialized_record)
+
+            # yield batch if we're at desired size
+            # -1 due to first row being field names
+            if len(batch) - 1 == n:
+                yield batch
+                batch = [serialized_csv_fields]
+                current_chars = len_csv_fields
+
+        # give back anything leftover
+        if len(batch) > 1:
+            yield batch
+
+    def _serialize_csv_record(self, record):
+        """Given a list of strings (record) return
+        the corresponding record serialized in .csv format"""
+        self.csv_writer.writerow(record)
+        self.csv_buff.seek(0)
+        serialized = self.csv_buff.read().encode("utf-8")
```

Comment:
I think you can just do this:
```suggestion
        serialized = self.csv_buff.getvalue().encode("utf-8")
```